packages/plugin/vite/src/config/vite.main.config.ts

```diff
@@ -1,21 +1,15 @@
-import { defineConfig, mergeConfig } from 'vite';
-import {
-  getBuildConfig,
-  getBuildDefine,
-  external,
-  pluginHotRestart,
-} from './vite.base.config.mjs';
+// eslint-disable-next-line node/no-unpublished-import
+import { type ConfigEnv, mergeConfig, type UserConfig } from 'vite';
 
-// https://vitejs.dev/config
-export default defineConfig((env) => {
-  /** @type {import('vite').ConfigEnv<'build'>} */
-  const forgeEnv = env;
+import { external, getBuildConfig, getBuildDefine, pluginHotRestart } from './vite.base.config';
+
+export function getConfig(forgeEnv: ConfigEnv<'build'>): UserConfig {
   const { forgeConfigSelf } = forgeEnv;
   const define = getBuildDefine(forgeEnv);
-  const config = {
+  const config: UserConfig = {
     build: {
       lib: {
-        entry: forgeConfigSelf.entry,
+        entry: forgeConfigSelf.entry!,
```

Comment:
1. Does `forgeConfigSelf` mean that this config is self-hosted by forge? Where is it?
2. Why use `entry!` here? I checked the call chain and found that there is a chance that entry could be `undefined`.

Reply:
Fixed in [eb12ad7](https://github.com/electron/forge/pull/3583/commits/eb12ad77007920409e96f1bfe0c05097c6bda98c)